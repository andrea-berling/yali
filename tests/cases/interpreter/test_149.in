class Wizard {
  getSpellCaster() {
    fun castSpell() {
      print this;
      print "Casting spell as " + this.name;
    }
    // Functions are first-class objects in Lox
    return castSpell;
  }
}
var wizard = Wizard();
wizard.name = "Merlin";
// Calling an instance method that returns a<|SPACE|>// function should work
wizard.getSpellCaster()();

