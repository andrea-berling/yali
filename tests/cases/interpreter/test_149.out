Wizard instance
Casting spell as Merlin
